src/modules/python/pbs_hooks/PBS_power.PY

```diff
@@ -0,0 +1,191 @@
+        me = pbs.get_local_nodename()
+        this_event.vnode_list[me].resources_available[
+            "eoe"] = power._map_profile_names(profiles)
+    power.disconnect()
+    this_event.accept()
+
+
+# Gather energy usage for all jobs
+if this_event.type == pbs.EXECHOST_PERIODIC:
+    # Check if any jobs are running
+    if len(this_event.job_list) == 0:
+        this_event.accept()
+
+    power = init_power(this_event)
+    for jobid in this_event.job_list.keys():
+        # set energy usage
+        job = this_event.job_list[jobid]
+        if not job.in_ms_mom():		# skip any jobs that MOM is not MS
+            continue
+        if not vnodes_enabled(job):	# skip if vnodes have power_provisioning=0
+            continue
+
+        try:
+            usage = power.get_usage(job)
+            if usage is not None:
+                job.resources_used["energy"] = usage
+        except Exception, e:
+            pbs.logmsg(pbs.LOG_ERROR, str(e))
+    power.disconnect()
+    this_event.accept()
+
+# From this point on, the event will have a job.
+this_job = this_event.job
+
+# The following code is needed until spid 329611 is fixed.
+if this_event.type == pbs.EXECJOB_BEGIN:
+    me = pbs.get_local_nodename()
+    if not _svr_vnode(me).power_provisioning:
+        this_event.accept()
+    requested_profile = str(this_job.schedselect).partition(
+        'eoe=')[2].partition('+')[0].partition(':')[0]
+    if requested_profile != "":
+        try:
+            this_event.vnode_list[me].current_eoe = requested_profile
+        except:
+            pass
+    this_event.accept()
+if this_event.type == pbs.EXECJOB_END:
+    me = pbs.get_local_nodename()
+    try:
+        this_event.vnode_list[me].current_eoe = None
+    except:
+        pass
+    this_event.accept()
+# End code needed for spid 329611.
+
+# No further processing is needed if we are not mother superior.
+if not this_job.in_ms_mom():
+    this_event.accept()
+
+if not vnodes_enabled(this_job):	# Don't do anything if power_provisioning=0
+    this_event.accept()
+
+if this_event.type == pbs.EXECJOB_PROLOGUE:
+    # Was an EOE requested?
+    requested_profile = str(this_job.schedselect).partition(
+        'eoe=')[2].partition('+')[0].partition(':')[0]
+
+    power = init_power(this_event)
+    try:
+        power.activate_profile(requested_profile, this_job)
+        power.disconnect()
+    except Exception, e:
+        power.disconnect()
+        this_event.reject(str(e))
+elif this_event.type == pbs.EXECJOB_EPILOGUE:
+    power = init_power(this_event)
+    # set energy usage
+    try:
+        usage = power.get_usage(this_job)
+        if usage is not None:
+            this_job.resources_used["energy"] = usage
+    except Exception, e:
+        pbs.logjobmsg(this_job.id, str(e))
+
+    try:
+        power.deactivate_profile(this_job)
+    except Exception, e:
+        pbs.logjobmsg(this_job.id, str(e))
+    power.disconnect()
+else:
+    this_event.reject("Unexpected event type")
```

Comment:
you can probably remove this block. You are already checking for event-type and accepting the event earlier.